docs/site/tutorials/connectors/Mysql-tutorial.md

```diff
@@ -0,0 +1,299 @@
+  port: 3306,
+  user: 'loopback',
+  password: 'pa55w0rd',
+  database: 'demo',
+};
+```
+
+### 4. Create repositories
+
+A [Repository](../../Repository.md) is an artifact that ties the model and the
+datasource. We will need to create the repository for the `User` class to access
+the database. The steps of creating `UserRepository` by running `lb4 repository`
+are skipped here:
+
+{% include code-caption.html content="user.repository.ts" %}
+
+```ts
+// imports
+export class UserRepository extends DefaultCrudRepository<
+  User,
+  typeof User.prototype.id,
+  UserRelations
+> {
+  constructor(@inject('datasources.db') dataSource: DbDataSource) {
+    super(User, dataSource);
+  }
+}
+```
+
+### 5. Database migration
+
+LoopBack provides a convenient way to create schemas/tables for our models if we
+don't have corresponding schemas defined in the relational database. Once we
+created the above artifacts, run the following commands:
+
+**1.** Build the project:
+
+```bash
+$ npm run build
+```
+
+**2.** Migrate database schemas (alter existing tables):
+
+```bash
+$ npm run migrate
+```
+
+This would generate the corresponding MySQL table `User` using the metadata from
+`User` model via Auto-migrate if it does not exist. See
+[Database Migrate](#database-migration) section below for information.
+
+If you check the database, you should able to see the table `User`.
+
+```
+mysql> describe User;
++------------+--------------+------+-----+---------+----------------+
+| Field      | Type         | Null | Key | Default | Extra          |
++------------+--------------+------+-----+---------+----------------+
+| id         | int(11)      | NO   | PRI | NULL    | auto_increment |
+| name       | varchar(512) | YES  |     | NULL    |                |
+| hasAccount | tinyint(1)   | YES  |     | NULL    |                |
++------------+--------------+------+-----+---------+----------------+
+```
+
+### 6. Create endpoints and view data using API Explorer
+
+Once we built a [controller](../../Controllers.md) with `lb4 controller` to
+handle requests:
+
+```bash
+$ lb4 controller
+? Controller class name: user
+Controller User will be created in src/controllers/user.controller.ts
+
+? What kind of controller would you like to generate? REST Controller with CRUD
+functions
+? What is the name of the model to use with this CRUD repository? User
+? What is the name of your CRUD repository? UserRepository
+? What is the name of ID property? id
+? What is the type of your ID? number
+? Is the id omitted when creating a new instance? Yes    <-- id will be auto-generated
+? What is the base HTTP path name of the CRUD operations? /users
+```
+
+From the project root, start the app:
+
+```bash
+$ npm start
+```
+
+We can verify what we just created with API Explorer
+[`http://localhost:3000/explorer/`](http://localhost:3000/explorer/).
+
+## Database Migration
+
+[Database migration](../../Database-migrations.md) helps you create relational
```

Comment:
(I thought I submitted the comment but didn't see it. :) )
There's a database migration in the above section, do you want to move this paragraph up there?

Also, there are 2 things to note, see if you want to incorporate (needs rewording though):
1. if you make further changes, make sure to run `npm run build` before running the migrate script again, otherwise the changes won't take effect.
2. if you run the migrate script against an existing table, all the data will be lost.  See https://loopback.io/doc/en/lb4/Database-migrations.html#overview.

Reply:
I'd keep them separated so that users won't be overwhelmed when following the tutorial. I will reword the second part to make it more consistent. 
>you make further changes, make sure to run npm run build before running the migrate script again, otherwise the changes won't take effect.

Good idea.

> if you run the migrate script against an existing table

I think `npm run migrate -- --rebuild` drops the tables and `npm run migrate` just does the updating? Anyway I think it's worth mentioning it 👍 thanks!

Reply:
You might want to double check. IIRC, "migrate" always drop, and "update" will keep the data.

Reply:
I checked the script in the LB4 app: 
```
async function migrate(args) {
    const existingSchema = args.includes('--rebuild') ? 'drop' : 'alter';
```
And in repository package we have 
```
    async migrateSchema(options: SchemaMigrationOptions = {}): Promise<void> {
      const operation =
        options.existingSchema === 'drop' ? 'automigrate' : 'autoupdate';
```
I think if we don't have the option `--rebuild` it defaults to autoupdate.

Reply:
Yeah https://loopback.io/doc/en/lb4/Database-migrations.html#auto-update-the-database-explicitly mentions that by default it alters and `--rebuild` drops 👍